### PR TITLE
 Chore: refactor design smells

### PR DESCRIPTION
**Pull request by Patel-Raj**
I have done a few code refactors to reduce design smells and increase readability. This is part of my University assignment.
These are in continuation to the merged [PR](https://github.com/jenkinsci/opentelemetry-plugin/pull/754#event-10996368833).

- [Pull up methods](https://refactoring.guru/pull-up-method) - Commit: [1f2b267](https://github.com/jenkinsci/opentelemetry-plugin/commit/1f2b2677e1fe3d4b4c351f92e2261697260ce3c2)

Moved hashCode method to super class.

- [Move Method](https://refactoring.guru/move-method) - Commit: [e900b38](https://github.com/jenkinsci/opentelemetry-plugin/commit/e900b38cb41f51590a1d81b67b39035df78a8819)

Moved getLastExportedMetricByMetricName() method to JenkinsOtelPluginIntegrationTest class as it was used at only one place, to remove unnecessary abstraction smell.

- [Extract class](https://refactoring.guru/extract-class) - Commit: [4357725](https://github.com/jenkinsci/opentelemetry-plugin/commit/4357725b1954d173bc4dc8f51be15af157ad6f23)

Moved all static final constants to OtelConstantUtils class to make constants more cohesive.

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/OtelUtils.java [hunk 1]
```diff
@@ -51,21 +51,10 @@
...
-    public static final String MULTIBRANCH = "multibranch";
-    public static final String WORKFLOW = "workflow";
-    public static final String UNKNOWN = "unknown";
-    public static final String BRANCH = "branch";
-    public static final String CHANGE_REQUEST = "change_request";
-    public static final String TAG = "tag";
-    public static final String JENKINS_CORE = "jenkins-core";
-    public static final String UNKNOWN_VALUE = "#unknown";
-
     @CheckForNull
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/OtelUtils.java:58]: I do not see why move it from its place

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java [hunk 2]
```diff
@@ -217,4 +217,9 @@ public int compareTo(ObservabilityBackendDescriptor o) {
             throw new IllegalStateException("failure to create error template");
         }
     }
+
+    // pulled up common method to super class.
+    public int generateHashCode(Object... objects) {
```
Comment (kuisathaverat) [inline src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java:222]: this HashCode implementation is too general, you lose the control over the hascode generator and will ends in a error-prone hashcode generator, so a wrong match on equals comparisons.

**Review by kuisathaverat — CHANGES_REQUESTED** [review]

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java (hunk 2 above)
Comment (cyrille-leclerc) [reply src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java:222]: Same as @kuisathaverat , I don't perceive this change as a beautification of our code.

**Review by cyrille-leclerc — CHANGES_REQUESTED** [review]
Same as @kuisathaverat , thanks for the help but I feel several changes don't beautify our code.

File: src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java (hunk 2 above)
Comment (Patel-Raj) [reply src/main/java/io/jenkins/plugins/opentelemetry/backend/ObservabilityBackend.java:222]: Thanks for reviewing my changes. I see the reason why these refactorings do not help. I am closing the pull request.

**Pull request closed without merging**